### PR TITLE
Replace padding by stride

### LINKED ISSUE #13: Change "padding" to "stride"
(_Original issue posted by @hartwiganzt on private repo_)

Now, that we already have dense, ELL matrices, I conclude from a discussion with Mark that the name "padding" is probably inappropriate. Padding is what you add to make it fit nicely. What we use it for is something different. Traditionally called "leading dimension" Mark suggests the name "stride".

```
> *--*            \
> ||||            |
> ||||            | 
> ||||            | stride
> *--* \          |
> :  : | padding  |
> *--* /          /
```

_EDIT_: In more detail, the extra amount of memory added to a multidimensional array (unfolded as a series of vectors of size `n`) to assure aligned access to each vector is usually called `padding`. Meaning that the distance (usually called `stride`) between first elements of two consecutive vectors is exactly `stride = n + padding`.

Current Ginkgo naming is a bit misleading as it calls the distance between two first elements `padding` (instead of `stride`).

- [ ] replace all references to `padding` with `stride`

### PR DESCRIPTION
**Pull request by yanjen**
Replace "padding" with "stride".
This closes #13

### CODE REVIEW

**Comment by yanjen** [comment]
I think users can get padding simple by minus stride by the number of rows, also, padding is not actually used in the code.
We might not need this function.

**Comment by gflegar** [comment]
@hartwiganzt  I agree with @yanjen, I don't see a reason for adding the `get_padding()` function. I think most of the time user's will need `stride`, and for rare cases when they actually need the `padding` they can get it as @yanjen described.

We can always reconsider if at some point we see that `padding` gets used a lot (for now we don't have a single use for it in the library).

**Comment by yanjen** [comment]
@gflegar this also applies to the block jacobi preconditioner, right?

**Comment by gflegar** [comment]
@yanjen, yes, you are correct. Also change it in block-Jacobi

**Comment by hartwiganzt** [comment]
I think there are few places still having padding:
```
scc-wkitx-cl-cn-192-129 public_ginkgo/ginkgo> grep -r 'padding' core/*
core/base/lin_op.hpp:     * A->get_num_stored_elements() == A->get_num_rows() * A->get_padding()
core/matrix/dense.cpp:    const auto padding =
core/matrix/dense.cpp:                                                 max_nnz_per_row, padding);
scc-wkitx-cl-cn-192-129 public_ginkgo/ginkgo> grep -r 'padding' cpu/*
cpu/preconditioner/block_jacobi_kernels.cpp:              size_type num_blocks, uint32 max_block_size, size_type padding,
cpu/preconditioner/block_jacobi_kernels.cpp:           uint32 max_block_size, size_type padding,
cpu/preconditioner/block_jacobi_kernels.cpp:                  uint32 max_block_size, size_type padding,
cpu/preconditioner/block_jacobi_kernels.cpp:                      const Array<ValueType> &blocks, size_type block_padding,
cpu/preconditioner/block_jacobi_kernels.cpp:                      size_type result_padding) NOT_IMPLEMENTED;
scc-wkitx-cl-cn-192-129 public_ginkgo/ginkgo> grep -r 'padding' gpu/*
gpu/components/warp_blas.cuh: * @param padding  offset between two consecutive rows of the matrix
gpu/components/warp_blas.cuh:    ValueType *__restrict__ destination, size_type padding)
gpu/components/warp_blas.cuh:            destination[idx * padding + row_perm] = source_row[i * increment];

```

File: core/matrix/dense.hpp [hunk 1]
```diff
@@ -250,15 +250,15 @@ class Dense : public BasicLinOp<Dense<ValueType>>,
      * @param exec  Executor associated to the matrix
      * @param num_rows  number of rows
      * @param num_cols  number of columns
-     * @param padding  padding of the rows (i.e. offset between the first
+     * @param stride  stride of the rows (i.e. offset between the first
```
Comment (hartwiganzt) [inline core/matrix/dense.hpp]: this should still be padding in the comments.

**Comment by hartwiganzt** [comment]
Please resolve the merge conflicts, then we can merge.

**Comment by yanjen** [comment]
Ok, I am on it.

**Pull request merged**